### ISSUE TITLE
Uncaught TypeError: Cannot read property 'isComplete' of undefined

### ISSUE THREAD

**Issue by kuwala** [t9md/atom-vim-mode-plus#282]
Throws an error when some invalid vim commands are pressed in an empty file.

Steps to reproduce:

In an empty file
1. Press v to enter Visual Char Mode
2. Press r to enter Replace Mode

**Atom Version**: 1.7.4
**System**: Mac OS X 10.11.4
**Thrown From**: [vim-mode-plus](https://github.com/t9md/atom-vim-mode-plus) package, v0.42.0

### Stack Trace

Uncaught TypeError: Cannot read property 'isComplete' of undefined

```
At /Users/Daniel/.atom/packages/vim-mode-plus/lib/operation-stack.coffee:53

TypeError: Cannot read property 'isComplete' of undefined
    at OperationStack.process (/Users/Daniel/.atom/packages/vim-mode-plus/lib/operation-stack.coffee:65:20)
    at OperationStack.run (/Users/Daniel/.atom/packages/vim-mode-plus/lib/operation-stack.coffee:46:8)
    at Function.Base.run (/Users/Daniel/.atom/packages/vim-mode-plus/lib/base.coffee:266:31)
    at atom-text-editor.<anonymous> (/Users/Daniel/.atom/packages/vim-mode-plus/lib/base.coffee:260:74)
    at CommandRegistry.module.exports.CommandRegistry.handleCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:260:29)
    at /Applications/Atom.app/Contents/Resources/app.asar/src/command-registry.js:3:61
    at KeymapManager.module.exports.KeymapManager.dispatchCommandEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:580:16)
    at KeymapManager.module.exports.KeymapManager.handleKeyboardEvent (/Applications/Atom.app/Contents/Resources/app.asar/node_modules/atom-keymap/lib/keymap-manager.js:388:22)
    at WindowEventHandler.module.exports.WindowEventHandler.handleDocumentKeyEvent (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:98:36)
    at HTMLDocument.<anonymous> (/Applications/Atom.app/Contents/Resources/app.asar/src/window-event-handler.js:3:61)
```

### Commands

```
  3x -0:12.4.0 blur (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
     -0:10.4.0 vim-mode-plus:activate-characterwise-visual-mode (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
     -0:09.7.0 vim-mode-plus:replace (atom-text-editor.editor.vim-mode-plus.is-focused.visual-mode.characterwise)
     -0:09.7.0 blur (atom-text-editor.editor.vim-mode-plus.is-focused.visual-mode.characterwise)
     -0:08.4.0 editor:consolidate-selections (atom-text-editor.editor.vim-mode-plus-input.mini.is-focused)
     -0:08.4.0 core:cancel (atom-text-editor.editor.vim-mode-plus-input.mini.is-focused)
  2x -0:08.4.0 blur (atom-text-editor.editor.vim-mode-plus-input.mini.is-focused)
     -0:07.0 vim-mode-plus:activate-linewise-visual-mode (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
     -0:06.2.0 vim-mode-plus:replace (atom-text-editor.editor.vim-mode-plus.is-focused.visual-mode.linewise)
     -0:06.2.0 blur (atom-text-editor.editor.vim-mode-plus.is-focused.visual-mode.linewise)
     -0:03.9.0 editor:consolidate-selections (atom-text-editor.editor.vim-mode-plus-input.mini.is-focused)
     -0:03.9.0 core:cancel (atom-text-editor.editor.vim-mode-plus-input.mini.is-focused)
  2x -0:03.9.0 blur (atom-text-editor.editor.vim-mode-plus-input.mini.is-focused)
     -0:00.9.0 vim-mode-plus:activate-characterwise-visual-mode (atom-text-editor.editor.vim-mode-plus.normal-mode.is-focused)
     -0:00.6.0 vim-mode-plus:replace (atom-text-editor.editor.vim-mode-plus.is-focused.visual-mode.characterwise)
     -0:00.6.0 blur (atom-text-editor.editor.vim-mode-plus.is-focused.visual-mode.characterwise)
```

### Config

```json
{
  "core": {
    "disabledPackages": [
      "vim-mode"
    ]
  }
}
```

### Installed Packages

```coffee
# User
open-in-browsers, v0.0.15
vim-mode-plus, v0.42.0

# Dev
No dev packages
```

**Comment by t9md** [comment]
## NOTE for future reference

I investigated why this was caused, but not 100% clear.
Fix itself is not bad, but my understanding what was happening in this bug is obscure.

When `Replace` operator is executed in `vC` mode, [this line](https://github.com/t9md/atom-vim-mode-plus/blob/052d35a7b1742f556e55c4f4c51478c853a02ace/lib/input.coffee#L62) cause unintended mode change from `vC` to `normal`, not sure why, I can't find why this line cause mode change.
This auto-unintended-mode-change is happen only in empty buffer.

Basically mode change should not happen in the middle of operation execution, this cause code path goes unexpected flow.

To avoid this odd situation, I can simply disallow `v` for empty buffer.
Simple, but I couldn't find other solution at least from quick investigation.

```
Atom    : 1.7.4
Electron: 0.36.8
Chrome  : 47.0.2526.110
Node    : 5.1.1
```

**Comment by vinicius0026** [comment]
@t9md This error is also occurring when I try to replace a single quote ' with double quotes " (or vice versa) using replace character command (r).
 
Stack Trace:
```
Uncaught TypeError: Cannot read property 'isComplete' of undefined
    at OperationStack.process (/Users/vinicius/.atom/packages/vim-mode-plus/lib/operation-stack.js:180:13)
    at ReplaceCharacter.processOperation (/Users/vinicius/.atom/packages/vim-mode-plus/lib/base.js:119:34)
    at options.onConfirm (/Users/vinicius/.atom/packages/vim-mode-plus/lib/base.js:134:14)
    at confirm (/Users/vinicius/.atom/packages/vim-mode-plus/lib/focus-input.js:37:5)
    at editor.buffer.onDidChangeText (/Users/vinicius/.atom/packages/vim-mode-plus/lib/focus-input.js:54:7)
    at Function.module.exports.Emitter.simpleDispatch (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/node_modules/event-kit/lib/emitter.js:25:20)
    at Emitter.module.exports.Emitter.emit (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/node_modules/event-kit/lib/emitter.js:141:34)
    at TextBuffer.module.exports.TextBuffer.emitDidChangeTextEvent (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/text-buffer.js:1675:28)
    at TextBuffer.module.exports.TextBuffer.transact (/Applications/Atom.app/Contents/Resources/app/node_modules/text-buffer/lib/text-buffer.js:975:18)
    at TextEditor.module.exports.TextEditor.transact (/Applications/Atom.app/Contents/Resources/app/src/text-editor.js:1673:32)
    at /Applications/Atom.app/Contents/Resources/app/src/text-editor.js:1234:30
    at TextEditor.module.exports.TextEditor.mergeSelections (/Applications/Atom.app/Contents/Resources/app/src/text-editor.js:2621:24)
    at TextEditor.module.exports.TextEditor.mergeIntersectingSelections (/Applications/Atom.app/Contents/Resources/app/src/text-editor.js:2583:41)
    at TextEditor.module.exports.TextEditor.mutateSelectedText (/Applications/Atom.app/Contents/Resources/app/src/text-editor.js:1232:25)
    at TextEditor.module.exports.TextEditor.insertText (/Applications/Atom.app/Contents/Resources/app/src/text-editor.js:1198:25)
    at get_process.nextTick (/Applications/Atom.app/Contents/Resources/app/src/text-editor-component.js:1715:34)
    at _combinedTickCallback (internal/process/next_tick.js:67:7)
    at process._tickCallback (internal/process/next_tick.js:98:9)
```
```
Atom    : 1.21.1
Electron: 1.6.15
Chrome  : 56.0.2924.87
Node    : 7.4.0
vim-mode-plus: 1.11.2
```

**Comment by t9md** [comment]
@vinicius0026 you can reproduce it always when you do `r '` or `r "`. I cannot reproduce it.
And have no idea why the situation where this issue happens.

**Comment by vinicius0026** [comment]
@t9md It happens when the cursor is over a `"` or `'` or \` characters and then I do `r '` or `r "`. If the cursor is over other characters, the replacement works (I didn't test every possibility though).
Maybe it has something to do with this: my keyboard is configured as USA - International so when I press `'`, for example, and then press space, only the `'` character is outputed, the space is "swallowed". I guess this happens because there are characters that are formed by the combination of `'` and other letters, such as `ç = ' + c` or `é = ' + e`.
